ArcGISRuntimeSDKQt_CppSamples/Layers/BrowseOGCAPIFeatureService/BrowseOGCAPIFeatureService.cpp

```diff
@@ -0,0 +1,221 @@
+    m_collectionInfo = m_serviceInfo->featureCollectionInfos();
+
+    createFeatureCollectionList();
+}
+
+void BrowseOGCAPIFeatureService::createFeatureCollectionList()
+{
+    m_featureCollectionList.clear();
+
+    // Populate list with names of each feature collection
+    for (OgcFeatureCollectionInfo* info : m_collectionInfo)
+    {
+        m_featureCollectionList.push_back(info->title());
+    }
+
+    emit featureCollectionListChanged();
+}
+
+MapQuickView* BrowseOGCAPIFeatureService::mapView() const
+{
+    return m_mapView;
+}
+
+// Set the view (created in QML)
+void BrowseOGCAPIFeatureService::setMapView(MapQuickView* mapView)
+{
+    if (!mapView || mapView == m_mapView)
+        return;
+
+    m_mapView = mapView;
+    m_mapView->setMap(m_map);
+
+    emit mapViewChanged();
+}
+
+QUrl BrowseOGCAPIFeatureService::featureServiceUrl() const
+{
+    return m_featureServiceUrl;
+}
+
+QStringList BrowseOGCAPIFeatureService::featureCollectionList() const
+{
+    return m_featureCollectionList;
+}
+
+QString BrowseOGCAPIFeatureService::errorMessage() const
+{
+    return m_errorMessage;
+}
+
+void BrowseOGCAPIFeatureService::loadService(QUrl urlFromInterface)
+{
+    m_featureServiceUrl = urlFromInterface;
+    emit urlChanged();
+    loadFeatureService(m_featureServiceUrl);
+}
+
+void BrowseOGCAPIFeatureService::loadFeatureCollection(int selectedFeature)
+{
+    // Create a CollectionInfo object from the selected collection
+    OgcFeatureCollectionInfo *info = m_collectionInfo[selectedFeature];
+
+    // Assign the CollectedInfo to the m_featureCollectionTable property
+    m_featureCollectionTable = new OgcFeatureCollectionTable(info);
+
+    // Set the feature request mode to manual (only manual is currently supported)
+    // In this mode you must manually populate the table - panning and zooming won't request features automatically
+    m_featureCollectionTable->setFeatureRequestMode(FeatureRequestMode::ManualCache);
+
+    // Populate the OGC feature collection table
+    QueryParameters queryParameters;
+    queryParameters.setMaxFeatures(1000);
+    m_featureCollectionTable->populateFromService(queryParameters, false, QStringList{ /* empty */ });
+
+    // Delete existing feature layer and create new layer from selected collection
+    delete m_featureLayer;
+    m_featureLayer = new FeatureLayer(m_featureCollectionTable);
+
+    // Connect loadStatusChanged with checkIfLayerLoaded
+    connect(m_featureLayer, &FeatureLayer::loadStatusChanged, this, &BrowseOGCAPIFeatureService::checkIfLayerLoaded);
+
+    // Connect errorOccurred to handleError
+    connect(m_featureLayer, &FeatureLayer::errorOccurred, this, &BrowseOGCAPIFeatureService::handleError);
+
+    // Load feature layer
+    m_featureLayer->load();
+}
+
+void BrowseOGCAPIFeatureService::checkIfLayerLoaded(LoadStatus loadStatus)
+{
+    if (loadStatus != LoadStatus::Loaded)
+    {
+        return;
+    }
+    addFeatureToMap();
```

Comment:
i think its a feature_Layer_ that you are adding to the map

Reply:
Yeah, a features are the individual geometries/attributes within a feature layer. There may be many feature layers in a feature service.